tests/test_arc32.py

```diff
@@ -471,6 +482,65 @@ def test_arc4_routing(
     app_client.call("method_with_default_args")
 
 
+def test_arc4_routing_with_many_params(
+    algod_client: AlgodClient,
+    account: algokit_utils.Account,
+    asset_a: int,
+    asset_b: int,
+) -> None:
+    app_spec = algokit_utils.ApplicationSpecification.from_json(
+        compile_arc32(TEST_CASES_DIR / "abi_routing")
+    )
+    app_client = algokit_utils.ApplicationClient(algod_client, app_spec, signer=account)
+
+    # create
+    create_response = app_client.create()
+    assert create_response.confirmed_round
+
+    result = app_client.call(
+        "method_with_more_than_15_args",
+        a=1,
```

Comment:
Could we pass different values for each arg just to verify things are unpacked as expected?

Maybe bytes b"a" - b"v" and concat for the result?